File: public/types/interview.ts
export interface InterviewSession {
  id: string;
  candidate_id: string;
  job_position: string;
  interview_type: 'practice' | 'real' | 'voice_only';
  status: 'scheduled' | 'in_progress' | 'completed';
}

export interface InterviewConfig {
  job_description: string;
  required_skills: string[];
  language: string;
  avatar_enabled: boolean;
}

export interface InterviewScore {
  overallScore: number;
  technicalScore: number;
  communicationScore: number;
  confidenceScore: number;
  sentimentAnalysis: Record<string, any>;
  voiceAnalysis: Record<string, any>;
}

export interface InterviewResult {
  sessionId: string;
  score: InterviewScore;
  transcript: string;
  feedback: string;
  recommendations: string[];
}

export interface WebRTCRoom {
  roomId: string;
  roomUrl: string;
  token: string;
}

File: server/app/models/interview.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from enum import Enum

class InterviewStatus(str, Enum):
    SCHEDULED = "scheduled"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"

class InterviewType(str, Enum):
    PRACTICE = "practice"
    REAL = "real"
    VOICE_ONLY = "voice_only"

class InterviewSession(BaseModel):
    id: Optional[str] = None
    candidate_id: str
    job_position: str
    interview_type: InterviewType
    status: InterviewStatus = InterviewStatus.SCHEDULED
    bot_url: Optional[str] = None
    
class InterviewConfig(BaseModel):
    job_description: str
    required_skills: List[str]
    language: str = "en"
    avatar_enabled: bool = True